src/main/java/org/roaringbitmap/longlong/RoaringTreeMap.java

```diff
@@ -0,0 +1,284 @@
+package org.roaringbitmap.longlong;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.roaringbitmap.IntIterator;
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+
+//  this class is not thread-safe
+//@Beta
+public class RoaringTreeMap {
+	protected final NavigableMap<Integer, MutableRoaringBitmap> hiToBitmap = new TreeMap<>();
```

Comment:
To reiterate: why are you using the the byte buffer implementation? You will see significantly better performance with RoaringBitmap as opposed to buffer.* classes. If you need to use buffer.* classes, you should move all of longlong underneath org.roaringbitmap.buffer.